substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrRuntimeAccessImpl.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.jfr;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.oracle.svm.jfr.traceid.JfrTraceIdMap;
+import jdk.internal.event.Event;
+
+class JfrRuntimeAccessImpl implements JfrRuntimeAccess {
+
+    private final List<Class<? extends Event>> eventClasses = new ArrayList<>();
+
+    private final Set<ClassLoader> reachableClassloaders = new HashSet<>();
```

Comment:
As the data structures referenced in `eventClasses` and `reachableClassloaders` end up in the image heap and therefore increase the size of the image, it would be better to use plain arrays instead of `ArrayList` and `HashSet`. One way of doing that would be that only the code in `JfrFeature` uses `ArrayList` and `HashSet` data structures. The data could then be convert to arrays before it is passed to the `JfrRuntimeAccessImpl` constructor.